app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -360,4 +360,42 @@ object BitcoindRpcBackendUtil extends Logging {
 
     resultF
   }
+
+  def startBitcoindMempoolPolling(
+      bitcoind: BitcoindRpcClient,
+      interval: FiniteDuration = 10.seconds)(
+      processTx: Transaction => Future[Unit])(implicit
+      system: ActorSystem,
+      ec: ExecutionContext): Cancellable = {
+    @volatile var prevMempool: Set[DoubleSha256DigestBE] =
+      Set.empty[DoubleSha256DigestBE]
+
+    def getDiffAndReplace(
+        newMempool: Set[DoubleSha256DigestBE]): Set[DoubleSha256DigestBE] = {
+      val txids =
+        if (prevMempool.isEmpty) Set.empty[DoubleSha256DigestBE]
+        else
+          newMempool.diff(prevMempool)
+      prevMempool = newMempool
+      txids
+    }
+
+    system.scheduler.scheduleWithFixedDelay(0.seconds, interval) { () =>
+      {
+        logger.trace("Polling bitcoind for mempool")
+
+        for {
+          mempool <- bitcoind.getRawMemPool
+          newTxIds = getDiffAndReplace(mempool.toSet)
+          _ = logger.trace(s"Found ${newTxIds} new mempool transactions")
+          newTxs <- FutureUtil.sequentially(newTxIds)(
+            bitcoind.getRawTransactionRaw(_))
+          _ <- FutureUtil.sequentially(newTxs)(processTx)
```

Comment:
may want to consider using an akka stream for this where the line above is a `Source[Transaction,NotUsed]` or something like that. That means we can start processing the transactions without having to wait for all the requests to bitcoind to complete.